app/src/main/kotlin/com/intive/picover/main/navigation/view/MainScreen.kt

```diff
@@ -23,90 +24,84 @@ import androidx.navigation.NavGraph.Companion.findStartDestination
 import androidx.navigation.NavHostController
 import androidx.navigation.compose.currentBackStackEntryAsState
 import androidx.navigation.compose.rememberNavController
-import com.google.accompanist.navigation.material.BottomSheetNavigator
 import com.google.accompanist.navigation.material.ExperimentalMaterialNavigationApi
 import com.google.accompanist.navigation.material.rememberBottomSheetNavigator
 import com.intive.picover.main.navigation.model.NavigationItem
 
-@OptIn(ExperimentalMaterial3WindowSizeClassApi::class, ExperimentalMaterialNavigationApi::class)
+@OptIn(ExperimentalMaterial3WindowSizeClassApi::class)
 @Composable
 fun MainScreen(activity: Activity, snackbarHostState: SnackbarHostState) {
 	val windowSize = calculateWindowSizeClass(activity)
-	val bottomSheetNavigator = rememberBottomSheetNavigator()
-	val navController = rememberNavController(bottomSheetNavigator)
-	val snackbarHost = @Composable {
-		SnackbarHost(snackbarHostState)
-	}
-	if (windowSize.widthSizeClass == WindowWidthSizeClass.Compact) {
-		BottomBarNavigation(navController, bottomSheetNavigator, snackbarHost)
-	} else {
-		RailNavigation(navController, bottomSheetNavigator, snackbarHost)
-	}
+	NavigationBar(
+		snackbarHostState = snackbarHostState,
+		isRail = windowSize.widthSizeClass != WindowWidthSizeClass.Compact,
+	)
 }
 
-@OptIn(ExperimentalMaterialNavigationApi::class)
 @Composable
-private fun BottomBarNavigation(
+fun RowScope.NavigationItems(
 	navController: NavHostController,
-	bottomSheetNavigator: BottomSheetNavigator,
-	snackbarHost: @Composable () -> Unit,
+	isRail: Boolean,
 ) {
-	Scaffold(
-		snackbarHost = snackbarHost,
-		bottomBar = {
-			NavigationBar(modifier = Modifier.fillMaxWidth()) {
-				val backStackEntry = navController.currentBackStackEntryAsState()
-				NavigationItem.entries.forEach { item ->
-					NavigationBarItem(
-						selected = item.route == backStackEntry.value?.destination?.route,
-						onClick = { navController.navigateWithSingleTop(item) },
-						icon = {
-							Icon(
-								imageVector = item.icon,
-								contentDescription = stringResource(id = item.labelResId),
-							)
-						},
-						label = {
-							Text(text = stringResource(id = item.labelResId))
-						},
-					)
-				}
-			}
-		},
-	) { innerPadding ->
-		PicoverNavHost(
-			modifier = Modifier.padding(innerPadding),
-			navController = navController,
-			bottomSheetNavigator = bottomSheetNavigator,
-		)
+	val backStackEntry = navController.currentBackStackEntryAsState()
+	NavigationItem.entries.forEach { item ->
+		val selected = item.route == backStackEntry.value?.destination?.route
+		val onClick = {
+			navController.navigateWithSingleTop(item)
+		}
+		val icon = @Composable {
```

Comment:
Could it be extracted to function instead of storing it in variable? In my opinion it will be easier to read this way.

Reply:
Done